src/v/kafka/server/partition_proxy.cc

```diff
@@ -30,9 +30,12 @@ std::optional<partition_proxy> make_partition_proxy(
         return std::nullopt;
     }
     auto source_ntp = model::ntp(ntp.ns, *mts, ntp.tp.partition);
+    auto src_log = pm.get(source_ntp);
+    vassert(src_log, "Materialized topic must have an associated source");
```

Comment:
does this assertion assume that multiple indexes (e.g. partition manager, metadata cache, etc..) are updated atomically? if so, is that true? if not, is this assertion going to hard-stop redpanda under some benign races?

Reply:
Reply to same comment in this commit applies here too